File: DataAlgo_forPy/Stacks_Queues/stack.rb
class Stack
  def initialize
    @items = []
  end

  def push(item)
    @items.push(item)
  end

  def pop
    @items.pop()
  end

  def is_empty?
    @items.empty?
  end

  def peek
    @items.last
  end

  def size
    @items.length
  end

end

myStack = Stack.new

p myStack.is_empty?

myStack.push(1)
myStack.push(2)
myStack.push("three")

p myStack.peek

p myStack.is_empty?

myStack.pop()
p myStack.pop()
myStack.pop()

p myStack.is_empty?

File: DataAlgo_forPy/Stacks_Queues/stack.js
class Stack {
  constructor() {
    this.items = []
  }

  isEmpty() {
    return this.items.length === 0
  }

  push(item) {
    this.items.push(item)
  }

  pop() {
    return this.items.pop()
  }

  peek() {
    return this.items[this.items.length - 1];
  }

  size() {
    return this.items.length
  }
}

const myStack = new Stack()

console.log(myStack.isEmpty())

myStack.push(1)
myStack.push("two")
myStack.push({ three: "three" })

console.log(myStack.peek())
console.log(myStack.size())

myStack.pop()
myStack.pop()
console.log(myStack.pop())

console.log(myStack.isEmpty())
